recipe/meta.yaml

```
@@ -11,40 +11,60 @@ source:
 
 build:
   number: 0
-  noarch: python
+  skip: True  # [py<37]
   entry_points:
     - pip-compile = piptools.scripts.compile:cli
     - pip-sync = piptools.scripts.sync:cli
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install --no-deps --no-build-isolation . -vv
 
 requirements:
   host:
-    - python >=3.7
+    - python
     - setuptools-scm
+    - setuptools
     - pip
+    - wheel
   run:
-    - python >=3.7
+    - python
     - python-build
-    - click >=7
-    - pip >=21.2
+    - click >=8
+    - pip >=22.2
     - setuptools
     - wheel
+    - tomli  # [py<311]
 
 test:
+  source_files:
+    - tests
   imports:
     - piptools
   commands:
     - pip check
     - pip-compile --help
     - pip-sync --help
```

```suggestion
    - pip-sync --help
    # Fix locale settings on osx
    - export LANGUAGE=en_US.UTF-8 # [osx]
    - export LC_ALL=en_US.UTF-8 # [osx]
    - export LANG=en_US.UTF-8 # [osx]
    - export LC_CTYPE=en_US.UTF-8 # [osx]
```
I think this would fix the test issues specific to osx.